src/stores/useWebSocketStore.tsx

```diff
@@ -0,0 +1,62 @@
+import { create } from 'zustand';
+import { combine } from 'zustand/middleware';
+
+import { Client, IMessage, StompSubscription } from '@stomp/stompjs';
+
+type State = {
+  isConnected: boolean;
+  topics: Map<string, StompSubscription>;
+  connect: () => void;
+  disconnect: () => void;
+  subscribe: (topic: string, callback: (message: IMessage) => void) => void;
+  unsubscribe: (topic: string) => void;
+};
+
+export const useWebSocketStore = create<State>(
+  combine(
+    {
+      isConnected: false,
+      topics: new Map<string, StompSubscription>(),
+    },
+    (set, get) => {
+      const client = new Client({
+        brokerURL: 'ws://localhost:8080/websocket',
+        onConnect: () => {
+          set({ isConnected: true });
+        },
+        onDisconnect: () => {
+          set({
+            isConnected: false,
+            topics: new Map<string, StompSubscription>(),
+          });
+        },
+      });
+
+      return {
+        connect: () => {
+          client.activate();
+        },
+        disconnect: () => {
```

Comment:
Please recreate new empty Map when client is disconnected

Reply:
we have it at the line 30 already, is that correct code?